frontend/appflowy_tauri/src/appflowy_app/components/database/grid/GridCell/GridSelectCell/GridSelectCell.tsx

```diff
@@ -0,0 +1,172 @@
+}> = ({ rowId, field, cell }) => {
+  const viewId = useViewId();
+  const options = cell?.data?.options ?? [];
+  const selectedIds = useMemo(() => cell?.data.selectOptions?.map(({ id }) => id )?? [], [cell?.data.selectOptions]);
+  const [newOptionName, setNewOptionName] = useState('');
+  const [domRef, setDomRef] = useState<HTMLElement | undefined>();
+  const [editingOption, setEditingOption] = useState<Database.SelectOption | undefined>();
+
+  const handleInput = useCallback((event: FormEvent) => {
+    const value = (event.target as HTMLInputElement).value;
+
+    setNewOptionName(value);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    setNewOptionName('');
+  }, []);
+
+  const handleChange = (event: SelectChangeEvent<string | string[]>) => {
+    const { target: { value } } = event;
+
+    const current = Array.isArray(value) ? value : [value];
+    const prev = cell?.data.selectOptions?.map(({ id }) => id);
+    const deleteOptionIds = prev?.filter(id => current.find(cur => cur === id) === undefined);
+
+    void service.updateSelectOptionCell(viewId, rowId, field.id, {
+      insertOptionIds: current,
+      deleteOptionIds,
+    });
+  };
+
+  const handleNewTagClick = async () => {
+    const exist = options.find(option => option.name.toLowerCase() === newOptionName.toLowerCase());
+
+    if (exist) {
+      return service.updateSelectOptionCell(viewId, rowId, field.id, {
+        insertOptionIds: [exist.id],
+      });
+    }
+
+    const option = await service.createSelectOption(viewId, field.id, newOptionName);
+
+    await service.insertOrUpdateSelectOption(viewId, field.id, [option], rowId);
+  };
+
+  const handleDetailsClick = useCallback((event: React.MouseEvent, option: Database.SelectOption) => {
+    event.stopPropagation();
+    setDomRef(event.target as HTMLElement);
+    setEditingOption(option);
+  }, []);
+
+  return (
+    <>
+      <Select
+        className="w-full"
+        classes={{
+          select: 'flex items-center gap-2 px-4 py-2 h-6',
+        }}
+        size="small"
+        value={selectedIds}
+        multiple={field.type === FieldType.MultiSelect}
+        input={<InputBase />}
+        IconComponent={() => null}
+        MenuProps={{
+          classes: {
+            list: 'py-5',
+          },
+          anchorOrigin: {
+            vertical: 'bottom',
+            horizontal: 'left',
+          },
+          transformOrigin: {
+            vertical: 'top',
+            horizontal: 'left',
+          },
+        }}
+        renderValue={(selected) => selected
+          .map((id) => options.find(option => option.id === id))
+          .map((option) => option && (
+            <Tag
+              key={option.id}
+              size="small"
+              color={option.color}
+              label={option.name}
+            />
+          ))}
+        onChange={handleChange}
+        onClose={handleClose}
+      >
+        <ListSubheader className="text-xs mb-4">
```

Comment:
How about extact the component to make it a bit cleaner and easier to manager.
```ts
const CustomListSubheader = ({ title, inputProps }) => (
  <ListSubheader className="text-xs mb-4">{title}</ListSubheader>
);

const CustomInputSubheader = ({ value, onInput }) => (
  <ListSubheader className="flex">
    <OutlinedInput size="small" value={value} onInput={onInput} />
  </ListSubheader>
);

<CustomListSubheader title={t('grid.selectOption.createTag')} />
<CustomInputSubheader value={newOptionName} onInput={handleInput} />

```

Reply:
Due to the change of the UI draft, there are some logical changes here.

And I think `ListSubheader` and `OutlinedInput` is simple enough so no need to extact components like `CustomListSubheader` or `CustomInputSubheader`.

For cleaner to read, I use clear variable `searchInput` to name the code segment.